packages/core/echo/indexing/src/index-schema.ts

```diff
@@ -50,14 +49,19 @@ export class IndexSchema extends Resource implements Index {
   }
 
   @trace.span({ showInBrowserTimeline: true })
-  async find(filter: Filter) {
-    if (filter.type?.itemId === undefined) {
+  async find(filter: IndexQuery) {
```

Comment:
The `update` method still requires reference to an object, so it is strange that in one place in the same class we pass Reference, and in another we use IndexQuery